python_modules/libraries/dagster-snowflake-pyspark/dagster_snowflake_pyspark/snowflake_pyspark_type_handler.py

```diff
@@ -18,13 +20,17 @@ def _get_snowflake_options(config, table_slice: TableSlice) -> Mapping[str, str]
         "Missing config: Warehouse is required when using PySpark with the Snowflake I/O manager.",
     )
 
+    # If the application is not set, use the default partner connection identifier
+    application = config.get("application", SNOWFLAKE_PARTNER_CONNECTION_IDENTIFIER_PYSPARK)
```

Comment:
debatable: we allow users to override this, but should we also enable them to disable this instead, say by passing False?

Reply:
whats the reason behind allowing this at all? i'd imagine users may want to turn this off, but why would they want to override it? (mostly asking for context as i'm not familiar with this config)

i think we should restrict the option for now and loosen it in the future, meaning, for now, we probably don't need to expose this as a config and just inject the constant value. if user asks for it, we can add options to disable or override.

Reply:
Given its OSS nature, I'd find it unfortunate if the community started to fork Dagster just to circumvent this. That's why I wanted to provide some form of an escape hatch. As to why they would do this, well I suppose perhaps a service company that specializes into data engineering solutions could entertain their own partnership with Snowflake.

Regarding disabling this entirely, I guess that's unnecessary. They can provide any random string and that would have the same effect.

I propose leaving this as is or to rip it off entirely. But I think since I already did the legwork leaving there is the best value.

Reply:
> Regarding disabling this entirely, I guess that's unnecessary.

Setting `application=None` in the constructor will set the value to `None` downstream - I guess this should be enough to "disable" the feature, if we decide to keep this as a config.

> i think we should restrict the option for now and loosen it in the future, meaning, for now, we probably don't need to expose this as a config and just inject the constant value. if user asks for it, we can add options to disable or override.

I agree with this. Adding `application` as a config so that users can override it is error-prone. 
- `None` can be passed in the current implementation, but can `None` be passed as a value in the options?
- How is the exception handled if the `value` passed to the constructor is invalid?

If we decide to keep `application` in the configs, I think we need better guardrails. At this stage, we could just remove it from the configs.

Reply:
after offline discussion, option to configure is stripped.